### PR TITLE
Resolve todo and add unit test

### PR DESCRIPTION
**Pull request by corneliusweig**
Related issue: #270

### CODE REVIEW

File: internal/testutil/tempdir.go [hunk 1]
```diff
@@ -72,3 +76,11 @@ func (td *TempDir) Write(file string, content []byte) *TempDir {
 	}
 	return td
 }
+
+func (td *TempDir) WritePlugin(file string, plugin index.Plugin) *TempDir {
```
Comment (corneliusweig) [inline internal/testutil/tempdir.go]: I figure this will be useful if we want to write more tests that involve the filesystem

Comment (ahmetb) [reply internal/testutil/tempdir.go]: eh, it's ok. i suspect we might forget to use it. :)

Comment (ahmetb) [reply internal/testutil/tempdir.go]: pls add `td.t.Helper()` and it's lgtm

Comment (corneliusweig) [reply internal/testutil/tempdir.go]: Done.

**Pull request merged**